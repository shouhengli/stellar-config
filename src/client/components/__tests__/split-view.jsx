import React from 'react';
import SplitView from '../split-view.jsx';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

describe('component split-view', () => {
  it('renders correctly when no selectedClass', () => {
    const component = shallow(
      <SplitView selectedClass={null} loadGraphSchemaContent={jest.fn()} />
    );
    expect(toJSON(component)).toMatchSnapshot();
  });

  it('renders correctly when selectedClass is given', () => {
    const component = shallow(
      <SplitView selectedClass={jest.fn()} loadGraphSchemaContent={jest.fn()} />
    );
    expect(toJSON(component)).toMatchSnapshot();
  });

  it('loadGraphSchemaContent is called on mount', () => {
    const mockLoadGraphSchemaContent = jest.fn();
    shallow(
      <SplitView
        selectedClass={null}
        loadGraphSchemaContent={mockLoadGraphSchemaContent}
      />
    );
    expect(mockLoadGraphSchemaContent).toHaveBeenCalled();
  });
});
