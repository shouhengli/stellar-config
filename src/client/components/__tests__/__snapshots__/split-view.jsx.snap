// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component split-view renders correctly when no selectedClass 1`] = `
<div
  className="split-view columns"
>
  <div
    className="view column is-2"
    key="leftChild"
  >
    <Connect(ClassList) />
  </div>
  <div
    className="view column"
    key="rightChild"
  >
    <Connect(Component) />
  </div>
</div>
`;

exports[`component split-view renders correctly when selectedClass is given 1`] = `
<div
  className="split-view columns"
>
  <div
    className="view column is-2"
    key="leftChild"
  >
    <Connect(ClassList) />
  </div>
  <div
    className="view column is-half"
    key="midChild"
  >
    <Connect(ClassEditor)
      loadGraphSchemaContent={[Function]}
    />
  </div>
  <div
    className="view column"
    key="rightChild"
  >
    <Connect(Component) />
  </div>
</div>
`;
